src/mpl/src/gpu/mpl_gpu_cuda.c

```diff
@@ -508,6 +508,11 @@ __attribute__ ((visibility("default")))
 CUresult CUDAAPI cuMemFree(CUdeviceptr dptr)
 {
     CUresult result;
+    if (!gpu_initialized) {
```

Comment:
It occurs to me that there could be a race condition here.

Reply:
Oof. So we need some lock or atomic scheme to protect the initialization...